src/main/kotlin/assertk/assertions/string.kt

```diff
@@ -19,7 +19,7 @@ fun Assert<String>.hasLineCount(lineCount: Int) {
  * @param ignoreCase true to compare ignoring case, the default if false.
  * @see [isNotEqualTo]
  */
-fun Assert<String>.isEqualTo(other: String, ignoreCase: Boolean = false) {
+fun Assert<String?>.isEqualTo(other: String?, ignoreCase: Boolean) {
```

Comment:
Is removing the default value a good idea? This causes the tests such as `                assert("").isNotEqualTo("test")` to invoke `Assert<T>.isNotEqualTo(expected: Any?)`.

Reply:
Hm, the extension method `String.isEqualTo(other, ignoreCase)` delegates to `equals()` so the behavior should be the same. But keeping the default would technically better match the signature.

Guess the question is if it's better to match the signature of the extension, or have `isEqualTo` have the same impl for all T.